src/components/balance/ModalTransferAmount.vue

```diff
@@ -184,31 +201,193 @@
 import IconBase from '@/components/icons/IconBase.vue';
-import IconSolidChevronDown from '@/components/icons/IconSolidChevronDown.vue';
+// import IconSolidChevronDown from '@/components/icons/IconSolidChevronDown.vue';
 import IconAccountSample from '@/components/icons/IconAccountSample.vue';
 import IconSolidSelector from '@/components/icons/IconSolidSelector.vue';
-import IconExchange from '@/components/icons/IconExchange.vue';
+// import IconExchange from '@/components/icons/IconExchange.vue';
 
 export default defineComponent({
   components: {
     ModalSelectAccountOption,
     IconBase,
-    IconSolidChevronDown,
+    // IconSolidChevronDown,
     IconAccountSample,
     IconSolidSelector,
-    IconExchange,
+    // IconExchange,
+    FormatBalance,
+  },
+  props: {
+    allAccounts: {
+      type: Array,
+      required: true,
+    },
+    allAccountNames: {
+      type: Array,
+      required: true,
+    },
+    address: {
+      type: String,
+      required: true,
+    },
+    accountIdx: {
+      type: Number,
+      required: true,
+    },
+    balance: {
+      type: BN,
+      required: true,
+    },
   },
   setup(props, { emit }) {
     const closeModal = () => {
       emit('update:is-open', false);
     };
 
+    const openOption = ref(false);
+
+    const unitToken = inject('unitToken', '');
+    const decimal = inject('decimal', 10);
+
+    const selAccount = ref(0);
+    const transferAmt = ref(0);
+
+    const toAccount = ref(props.allAccounts[0] as string);
+    const toAddress = ref(props.allAccounts[0] as string);
+    const toAccountName = ref(props.allAccountNames[0]);
+
+    const arrUnitNames = getUnitNames(unitToken);
+    const selectUnit = ref(unitToken);
+
+    // const { balance } = useBalance(defaultAccount);
+    // provide('balance', balance);
+
+    const formatBalance = computed(() => {
+      const tokenDecimal = decimal;
+      return plasmUtils.reduceBalanceToDenom(
+        props.balance.clone(),
+        tokenDecimal
+      );
+    });
+
+    const { api } = useApi();
+    const store = useStore();
+
+    const transfer = async (transferAmt: number, selAddress: string) => {
+      console.log('transfer', transferAmt);
+      console.log('selAccount', selAddress);
+      console.log('selUnit', selectUnit.value);
+
+      if (Number(transferAmt) === 0) {
+        console.log('The amount of token to be transmitted must not be zero');
+        store.dispatch(
+          ActionTypes.SHOW_ALERT_MSG,
+          `The amount of token to be transmitted must not be zero`
+        );
+        return;
+      }
+
+      try {
+        const unit = getUnit(selectUnit.value);
+        const toAmt = plasmUtils.reduceDenomToBalance(
+          transferAmt,
+          unit,
+          decimal
+        );
+        console.log('toAmt', toAmt.toString(10));
+
+        const injector = await web3FromSource('polkadot-js');
+        const transfer = await api?.value?.tx.balances.transfer(
+          selAddress,
```

Comment:
Please add a check to see if the provided address is correct before transferring any tokens.